### PR TITLE
Fixes for 4.13

### LINKED ISSUE #4086: Regressions in 4.13.2
## Issue Details

- [x] [Intersections_2](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.13.2-I-160/Intersections_2/TestReport_cgaltester_x86-64_Darwin-13.0_Apple-clang-5.0_Release.gz)
- [x] [Ipelets](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.13.2-I-160/CGAL_ipelets_Demo/TestReport_gimeno_Debian-Stable.gz)
- [Barycentric_coordinates_2_Examples](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.13.2-I-160/Barycentric_coordinates_2_Examples/TestReport_gimeno_Fedora-rawhide-Release.gz)  won't fix
- [x] [Segment Delaunay](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.13.2-I-160/Segment_Delaunay_graph_2/TestReport_lrineau_ArchLinux-clang-Release.gz)
- [x] [SDG again](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.13.2-I-160/Segment_Delaunay_graph_2/TestReport_lrineau_ArchLinux-clang-CXX17-Release.gz)
- [x] [Snap Rounding 2](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.13.2-I-160/Snap_rounding_2_Demo/TestReport_lrineau_ArchLinux-CXX17-Release.gz)
- [Triangulation_2](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.13.2-I-160/Triangulation_2/TestReport_lrineau_Debian-stable-Release.gz) won't fix.
- [Convex_hull_3_Examples](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.13.2-I-160/Convex_hull_3_Examples/TestReport_lrineau_Fedora-with-LEDA.gz)  won't fix

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes

warnings and errors fixes.
## Release Management
* Issue(s) solved (if any): fix #4086

### CODE REVIEW

File: Kernel_23/include/CGAL/determinant.h [hunk 1]
```diff
@@ -39,7 +39,7 @@ determinant(
  const RT& a10,  const RT& a11)
 {
 // First compute the det2x2
-  const RT m01 = a00*a11 - a10*a01;
+  const RT m01 = (a00*a11 - a10*a01);
```
Comment (lrineau) [inline Kernel_23/include/CGAL/determinant.h]: Strange. What was the warning, that an extra parenthesis can fix?

**Comment by lrineau** [comment]
This pull-request should have been a priority, and tested already two days ago. It is a blocker for CGAL-5.0-dev.

File: Kernel_23/include/CGAL/determinant.h (hunk 1 above)
Comment (maxGimeno) [reply Kernel_23/include/CGAL/determinant.h]: Can't find it again :/ Something about a wrong conversion between FT and RT but I don't think it fixed it.

**Comment by maxGimeno** [comment]
- There is a remaining issue that I cannot fix, I was waiting for some help with it before trying to integrate this.
- Those are all warnings that occur in 4.13 but not in 5.0, so I didn't see how it should be a blocker
- I was waiting for @sloriot to fix his PR about boost parameters to re-do a testsuite of 4.13.2
- @sloriot told me that fixing warnings for 4.13 was absolutely not a priority
- We tested and integrated lots of PRs for the 5.0 milestone, that required to test the master integration, and thus not the 4.13

For all those reasons, I did not make this PR a priority.

File: Kernel_23/include/CGAL/determinant.h (hunk 1 above)
Comment (lrineau) [inline Kernel_23/include/CGAL/determinant.h]:
```suggestion
  const RT m01 = RT(a00*a11 - a10*a01);
```

Comment (maxGimeno) [reply Kernel_23/include/CGAL/determinant.h]: That did not fix the error : 
```
/mnt/testsuite/include/CGAL/determinant.h:42:18: error: no matching conversion for C-style cast from '__gmp_expr<typename __gmp_resolve_expr<__mpq_struct [1], __mpq_struct [1]>::value_type, __gmp_binary_expr<__gmp_expr<__mpq_struct [1], __gmp_binary_expr<__gmp_expr<__mpq_struct [1], __gmp_binary_expr<__gmp_expr<mpq_t, mpq_t>, __gmp_expr<mpq_t, mpq_t>, __gmp_binary_minus> >, __gmp_expr<__mpq_struct [1], __gmp_binary_expr<__gmp_expr<mpq_t, mpq_t>, __gmp_expr<mpq_t, mpq_t>, __gmp_binary_minus> >, __gmp_binary_multiplies> >, __gmp_expr<__mpq_struct [1], __gmp_binary_expr<__gmp_expr<__mpq_struct [1], __gmp_binary_expr<__gmp_expr<mpq_t, mpq_t>, __gmp_expr<mpq_t, mpq_t>, __gmp_binary_minus> >, __gmp_expr<__mpq_struct [1], __gmp_binary_expr<__gmp_expr<mpq_t, mpq_t>, __gmp_expr<mpq_t, mpq_t>, __gmp_binary_minus> >, __gmp_binary_multiplies> >, __gmp_binary_minus> >' (aka '__gmp_expr<__mpq_struct [1], __gmp_binary_expr<__gmp_expr<__mpq_struct [1], __gmp_binary_expr<__gmp_expr<__mpq_struct [1], __gmp_binary_expr<__gmp_expr<mpq_t, mpq_t>, __gmp_expr<mpq_t, mpq_t>, __gmp_binary_minus> >, __gmp_expr<__mpq_struct [1], __gmp_binary_expr<__gmp_expr<mpq_t, mpq_t>, __gmp_expr<mpq_t, mpq_t>, __gmp_binary_minus> >, __gmp_binary_multiplies> >, __gmp_expr<__mpq_struct [1], __gmp_binary_expr<__gmp_expr<__mpq_struct [1], __gmp_binary_expr<__gmp_expr<mpq_t, mpq_t>, __gmp_expr<mpq_t, mpq_t>, __gmp_binary_minus> >, __gmp_expr<__mpq_struct [1], __gmp_binary_expr<__gmp_expr<mpq_t, mpq_t>, __gmp_expr<mpq_t, mpq_t>, __gmp_binary_minus> >, __gmp_binary_multiplies> >, __gmp_binary_minus> >') to '__gmp_expr<__mpq_struct [1], __gmp_binary_expr<__gmp_expr<mpq_t, mpq_t>, __gmp_expr<mpq_t, mpq_t>, __gmp_binary_minus> >'
  const RT m01 = (RT)(a00*a11 - a10*a01);
```

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-4.13.2-Ic-94.shtml

**Pull request merged**